File: src/Scripting/Lua/patterns/mvvm.lua
local model = { first = "Ada", last = "Lovelace" }
local viewmodel = {}
function viewmodel.full_name() return model.first .. " " .. model.last end
local view = function(vm) return "Hello " .. vm.full_name() end
assert(view(viewmodel) == "Hello Ada Lovelace")
return true

File: src/Scripting/PHP/patterns/mvvm.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('mvvm');
$model = ['first'=>'Ada','last'=>'Lovelace'];
$viewModel = new class($model) {
    public function __construct(private array $model) {}
    public function displayName(): string { return $this->model['first'].' '.$this->model['last']; }
};
$view = static fn($vm): string => $vm->displayName();
$check($view($viewModel) === 'Ada Lovelace');
